src/briefcase/commands/base.py

```diff
@@ -103,6 +104,26 @@ def split_passthrough(args):
         return args[:pos], args[pos + 1 :]
 
 
+def parse_config_overrides(config_overrides: list[str] | None) -> dict[str, Any]:
+    """Parse command line -C/--config option overrides.
+
+    :param config_overrides: The values passed in as configuration overrides. Each value
+        *should* be a "key=<valid TOML>" string.
+    :returns: A dictionary of app configuration keys to override and their new values.
+    :raises BriefcaseCommandError: if any of the values can't be parsed as valid TOML.
+    """
+    overrides = {}
+    if config_overrides:
+        for value in config_overrides:
+            try:
+                overrides.update(tomllib.loads(value))
```

Comment:
I can't see any mention of multi-level config values, like `--config 'linux.requires=...'`, which is [valid TOML](https://toml.io/en/v1.0.0#keys).

I think this code might actually work with that as long as you only had one of them. But then if you added `--config linux.something_else=...`, the second `linux` would overwrite the first instead of merging with it.

We don't necessarily need to support that syntax in this PR, but if we don't support it, we should at least display a useful error when the user tries to use it.

Reply:
Good call. Right now, it's probably easier to reject multi-level keys completely. We can always add them later if there's a reasonable use case.

Reply:
Actually - in the process of looking at this, it occurred to me that this approach avoids validation of any provided value... and when I went looking to see how I could do validation, I found a *much* cleaner way to handle the overrides, in a way that both validates *and* avoids the issue @rmartin16 found with not passing the overrides to some commands.